core/show_and_tell_model.py

```diff
@@ -45,7 +45,7 @@ def __init__(self, config, mode, train_inception=False):
           mode: "train", "eval" or "inference".
           train_inception: Whether the inception submodel variables are trainable.
         """
-        assert mode in ["train", "eval", "inference"]
+        assert mode in ["train", "eval", "inference"]  # nosec
```

Comment:
Seems like this is a real one? In optimized code this line would be gone and an arbitrary `mode` would get the model thrown off.

Reply:
I guess you're right. Not necessarily a security issue but I guess we should fix this.

Should I replace all the `asset`'s with `if` statements?

Reply:
Yeah I think so.